pennylane/ops/op_math/condition.py

```diff
@@ -501,9 +564,6 @@ def qnode(a, x, y, z):
 
         return cond_func
 
-    if qml.capture.enabled():
-        return _capture_cond(condition, true_fn, false_fn, elifs)
-
     if not isinstance(condition, MeasurementValue):

```

Comment:
The only concern I would've had would've been this line, which specifically avoids using `CondCallable` for MCM conditions. However, with capture enabled, these would be replaced with boolean tracers, so I believe we would still end up using `CondCallable`, and consequently, the conditional primitive, as expected.

I've integrated these changes with #6058 locally and can confirm that it works 😄

Reply:
nice! I was also worried about this line but this makes a lot of sense